vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt

```diff
@@ -67,11 +78,40 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
 
             // Schedule a timer to stop sharing
             scheduleTimer(roomArgs.roomId, roomArgs.durationMillis)
+
+            // Send beacon info state event
+            activeSessionHolder
+                    .getSafeActiveSession()
+                    ?.let { session ->
+                        session.coroutineScope.launch {
+                            sendBeaconInfo(session, roomArgs)
+                        }
+                    }
         }
 
         return START_STICKY
     }
 
+    private suspend fun sendBeaconInfo(session: Session, roomArgs: RoomArgs) {
+        val beaconContent = LiveLocationBeaconContent(
+                beaconInfo = BeaconInfo(
+                        timeout = roomArgs.durationMillis,
+                        isLive = true
+                ),
+                timestampAsMillisecond = clock.epochMillis()
+        ).toContent()
+
+        // This format is not yet finalized
+        val stateKey = session.myUserId
+        session
+                .getRoom(roomArgs.roomId)
+                ?.sendStateEvent(
+                        eventType = EventType.STATE_ROOM_BEACON_INFO,
```

Comment:
see the MCS: type should have the form `"m.beacon_info.@stefan:matrix.org"`
https://github.com/matrix-org/matrix-spec-proposals/blob/stefan/ephemeral-location-streaming/proposals/3672-ephemeral-location-streaming.md#proposal

Reply:
dynamic keys are quite painful for json parsers, do we have other examples of them in spec? would be great to avoid and replace with a content value if possible

Reply:
We also have https://github.com/matrix-org/matrix-spec-proposals/blob/andybalaam/owner-state-events/proposals/3757-restricting-who-can-overwrite-a-state-event.md but let me clarify this with the team.

Reply:
Updated.